libsolidity/interface/CompilerStack.cpp

```diff
@@ -397,7 +397,8 @@ bool CompilerStack::isRequestedContract(ContractDefinition const& _contract) con
 	return
 		m_requestedContractNames.empty() ||
 		m_requestedContractNames.count(_contract.fullyQualifiedName()) ||
-		m_requestedContractNames.count(_contract.name());
+		m_requestedContractNames.count(_contract.name()) ||
```

Comment:
Just from the context here, I don't understand why this change has the desired effect. Can you explain it?

Reply:
Oh, I just saw your comment. Of course: if a wildcard for the source name is given, `StandardCompiler::requestedContractNames()` will add the contract with the identifier `:C` instead of `A:C` (if source is `A`). This change adds an additionally checks for the contract name with the seperator `:` prefix. Another solution would have been to have the wildcard reflected with properly instead of implicitly accepting a whitespace, such as `_:A` or something like that.

Reply:
Is a name collision possible in that case?

Reply:
I don't think so, but I'm not 100% sure if I considered all use cases. Say, you would have 2 sources with the same contract, then the wildcard would result in both contracts being selected and compiled. Does that make sense to you?

Reply:
Ok, I see, you have to read that together with StandardCompiler.cpp:113